internal/httphandler/data_request_handler.go

```diff
@@ -176,7 +176,7 @@ func (d *dataRequestHandler) dataTransaction(response http.ResponseWriter, reque
 	}
 	if len(notSigned) > 0 {
 		sort.Strings(notSigned)
-		utils.SendHTTPResponse(response, http.StatusBadRequest,
+		utils.SendHTTPResponse(response, http.StatusUnauthorized,
```

Comment:
Can you add a unit test that covers that?

Reply:
There was a unit test that required the error code to be "bad request". I'm not sure why. And for some reason, the error code validation was commented out in the tests. I fixed the test and uncommented this check.